FinniversKit/Sources/Components/FrontPageShelfView/FrontPageShelfView.swift

```diff
@@ -0,0 +1,189 @@
+        
+        let section = NSCollectionLayoutSection(group: group)
+        section.contentInsets = NSDirectionalEdgeInsets(top: 0, leading: .spacingL, bottom: 0, trailing: 0)
+        section.orthogonalScrollingBehavior = .continuous
+        
+        
+        //Header
+        let headerSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .estimated(44))
+        let headerElement = NSCollectionLayoutBoundarySupplementaryItem(layoutSize: headerSize, elementKind: FrontPageShelfHeaderView.reuseIdentifier, alignment: .top)
+        section.boundarySupplementaryItems = [headerElement]
+        
+        return section
+    }
+    
+    private var savedSearchLayout: NSCollectionLayoutSection {
+        let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(90))
+        let item = NSCollectionLayoutItem(layoutSize: itemSize)
+        //Groups
+        let groupSize = NSCollectionLayoutSize(widthDimension: .absolute(80), heightDimension: .estimated(90))
+        let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitem: item, count: 1)
+        group.contentInsets = NSDirectionalEdgeInsets(top: 0, leading:0, bottom: 0, trailing: 16)
+        
+        //Sections
+        let section = NSCollectionLayoutSection(group: group)
+        section.contentInsets = NSDirectionalEdgeInsets(top: 0, leading: 20, bottom: 0, trailing: 0)
+        section.orthogonalScrollingBehavior = .continuous
+        
+        // Header
+        let headerSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .estimated(44))
+        let headerElement = NSCollectionLayoutBoundarySupplementaryItem(layoutSize: headerSize, elementKind: FrontPageShelfHeaderView.reuseIdentifier, alignment: .top)
+        section.boundarySupplementaryItems = [headerElement]
+        
+        return section
+    }
+}
+
+//MARK: - Datasource
+private extension FrontPageShelfView {
+    private func makeDatasource() -> Datasource {
+        let datasource = Datasource(collectionView: collectionView) { [weak self] collectionView, indexPath, item in
+            self?.shelfDatasource.frontPageShelfView(collectionView, cellForItemAt: indexPath, withItem: item)
+            
+        }
+        
+        datasource.supplementaryViewProvider = { (collectionView: UICollectionView, kind: String, indexPath: IndexPath) -> UICollectionReusableView? in
+            let headerView = collectionView.dequeue(FrontPageShelfHeaderView.self, for: indexPath, ofKind: kind)
+            let section = Section.allCases[indexPath.section]
+            switch section {
+            case .savedSearch:
+                if self.items[section, default: []].isEmpty { fallthrough }
+                headerView.configureHeaderView(withTitle: "Lagrede søk", buttonTitle: "Se alle", buttonAction: { print("Se alle lagrede søk")})
```

Comment:
We should configure these view models from the app instead (with localized Strings), instead of hardcoding in FinniversKit.